config/routes.rb

```diff
@@ -81,6 +81,10 @@
 
   get '/login', to: 'sessions#new'
   get '/logout', to: 'sessions#destroy'
+  
+  resources :csv, only: [ :index, :new, :create ]
```

Comment:
`resources :csvs, only: [:index, :new, :create, :show, :download]`

... maybe download can go away ... json/html = status and when requesting with .csv then it's a download ?